dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageForm.java

```diff
@@ -110,7 +112,24 @@ private TemplateLayout getTemplateLayout() throws BadRequestException {
 
             try {
                 String layoutString = JsonTransformer.mapper.writeValueAsString(layout);
-                return JsonTransformer.mapper.readValue(layoutString, TemplateLayout.class);
+                TemplateLayout layout = JsonTransformer.mapper.readValue(layoutString, TemplateLayout.class);
+
+                List<TemplateLayoutRow> rows = layout.getBody().getRows();
+                List<ContainerUUID> containers = rows.stream()
+                        .map(TemplateLayoutRow::getColumns)
+                        .flatMap(Collection::stream)
+                        .map(ContainerHolder::getContainers)
+                        .flatMap(Collection::stream)
+                        .filter(container -> !UtilMethods.isSet(container.getUUID()))
+                        .collect(Collectors.toList());
+
+                int nextContainerUUID = layout.getMaxContainerUUID() + 1;
+
+                for (ContainerUUID container : containers) {
+                    container.setUuid(String.valueOf(nextContainerUUID++));
```

Comment:
we should keep the immutable principle. Avoid if possible to modified the elements.